numba/ir_utils.py

```diff
@@ -691,33 +698,48 @@ def find_potential_aliases(blocks, args, typemap, func_ir, alias_map=None,
                 # only mutable types can alias
                 if is_immutable_type(lhs, typemap):
                     continue
-                if isinstance(expr, ir.Var) and lhs!=expr.name:
-                    _add_alias(lhs, expr.name, alias_map, arg_aliases)
-                # subarrays like A = B[0] for 2D B
-                if (isinstance(expr, ir.Expr) and (expr.op == 'cast' or
-                    expr.op in ['getitem', 'static_getitem'])):
-                    _add_alias(lhs, expr.value.name, alias_map, arg_aliases)
-                # array attributes like A.T
-                if (isinstance(expr, ir.Expr) and expr.op == 'getattr'
-                        and expr.attr in ['T', 'ctypes', 'flat']):
-                    _add_alias(lhs, expr.value.name, alias_map, arg_aliases)
-                # calls that can create aliases such as B = A.ravel()
-                if isinstance(expr, ir.Expr) and expr.op == 'call':
-                    fdef = guard(find_callname, func_ir, expr, typemap)
-                    # TODO: sometimes gufunc backend creates duplicate code
-                    # causing find_callname to fail. Example: test_argmax
-                    # ignored here since those cases don't create aliases
-                    # but should be fixed in general
-                    if fdef is None:
-                        continue
-                    fname, fmod = fdef
-                    if fdef in alias_func_extensions:
-                        alias_func = alias_func_extensions[fdef]
-                        alias_func(lhs, expr.args, alias_map, arg_aliases)
-                    if fmod == 'numpy' and fname in np_alias_funcs:
-                        _add_alias(lhs, expr.args[0].name, alias_map, arg_aliases)
-                    if isinstance(fmod, ir.Var) and fname in np_alias_funcs:
-                        _add_alias(lhs, fmod.name, alias_map, arg_aliases)
+                # Assignments like A = B alias.
+                if isinstance(expr, ir.Var):
+                    if lhs != expr.name:
+                        _add_alias(lhs, expr.name, alias_map, arg_aliases)
+                elif isinstance(expr, ir.Arg):
+                    if not expr.noalias:
+                        possibly_aliased_args.add(lhs)
+                    arg_aliases.add(lhs)
+                elif isinstance(expr, ir.Expr):
+                    # subarrays like A = B[0] for 2D B
```

Comment:
The following are assuming operations on numpy arrays.  It should check that the operand is indeed an array type.